doc/reference/std/errors.md

```diff
@@ -1,64 +1,421 @@
+
+### ContextError?
+```scheme
+(ContextError? obj)
+```
+
+Predicate testing whether the error is a context error
+
+### context-error?
+```scheme
+(def context-error? ContextError?)
+```
+
+Same as `ContextError?`.
+
+### KeyError
+```scheme
+(deferror-class KeyError)
+```
+
+Error indicating that some lookup operation failed because a key was unbound.
+
+### KeyError?
+```scheme
+(KeyError? obj)
+```
+
+Predicate testing whether the error is a key error
+
+### key-error?
+```scheme
+(def key-error? KeyError?)
+```
+
+Same as `KeyError?`.
+
+## Raising exceptions
+
+The library redefines `raise` and `error` so that it fills stack
+traces where appropriate.
+It also provides some utility procedure for raising particular errors
+without caring about the particulars of the condition system.
+
+### error
+```scheme
+(error message irritant ...)
+```
+
+Like basic Scheme's `error`, but raises with stack trace.
+
+### raise
+```scheme
+(raise exn)
+```
+
+Raises an exception condition.
+
+If the exception object is an instance of `StackTrace`, then unless
+there is already a continuation because this is a re-raise, the
+continuation is captured and stored in the `continuation` slot.
+
+### raise-bad-argument
+```scheme
+(raise-bad-argument where expectation bad)
+```
+
+Raises a `BadArgument` condition.
+
+### raise-io-error
+```scheme
+(raise-io-error where what . irritants)
+```
+
+Raises an `IOError` condition.
+
+### raise-premature-end-of-input
+```scheme
+(raise-premature-end-of-input where . irritants)
 ```
+
+Raises a `PrematureEndOfInput` condition.
+
+### raise-io-closed
+```scheme
+(raise-io-closed where what . irritants)
+```
+
+Raises an `IOClosed` condition.
+
+### raise-timeout
+```scheme
 (raise-timeout where what . irritants)
 ```
-:::
 
-Raises a `timeout-error`.
+Raises a `Timeout` condition.
+
+### raise-context-error
+```scheme
+(raise-context-error where what . irritants)
+```
+
+Raise a `ContextError` condition.
+
+### raise-key-error
+```scheme
+(raise-key-error where what . irritants)
+```
+
+Raise a `KeyError` condition.
+
+## BUGS
+
+Sometimes something that really shouldn't happen, but it did; because, Murphy.
+
+### BUG
+```scheme
+(BUG where what . irritants)
+```
+
+Raise a bug condition.
+
+### is-it-bug?
+```scheme
+(is-it-bug? obj)
+```
+
+Checks whether an error condition was caused by a something unexpected, a bug.
+
+## Defining custom Error classes
+
+The library offers a utility macro to define error classes that follow
+the standard library conventions. You don't have to use it, but if you
+are building a library it is recommended to do so.
+
+### deferror-class
+```scheme
+(deferror-class Class slots predicate-alias [constructor = Error:::init!])
+(deferror-class (Class Mixin ...) slots predicate-alias [constructor = Error:::init!])
+```
+
+The first form defines a class `Class` with slots `slots` that extends
+`Error`, mixing in `StackTrace`.  It also defines a predicate alias
```

Comment:
Wait, why isn't StackTrace already part of Error, then?

Reply:
ok, we can do that.